generate_embeddings.ipynb

```diff
@@ -0,0 +1,197 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "5ea23f97-50ba-4ddf-b856-a087e3ba3d55",
+   "metadata": {},
+   "source": [
+    "# Embeddings factory script\n",
+    "\n",
+    "Generate embeddings in bulk, on a per MGRS tile basis.\n",
+    "\n",
+    "Embeddings are stored into the `data/embeddings/` folder,\n",
+    "and then uploaded to `s3://clay-vector-embeddings/`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9b60152d-d504-4d59-b6de-d3a3a4526e2e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import glob\n",
+    "import os\n",
+    "import warnings\n",
+    "\n",
+    "import duckdb\n",
+    "import lightning as L\n",
+    "import torch\n",
+    "import tqdm\n",
+    "\n",
+    "from src.datamodule import ClayDataModule\n",
+    "from src.model_clay import CLAYModule"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "ae87a839-467c-4b18-9cf0-b540f9f1f6f1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set some environment variables and parameters\n",
+    "torch.set_float32_matmul_precision(precision=\"medium\")\n",
+    "os.environ[\"TORCH_CUDNN_V8_API_DISABLED\"] = \"1\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "95d01b40-f06d-4634-bbfb-dcb094361e46",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate list of MGRS tiles\n",
+    "#!aws s3 ls s3://clay-tiles-02/02/ | tr -s ' ' |  cut -d ' ' -f 3 | cut -d '/' -f 1 > mgrs_world.txt\n",
+    "mgrs_tiles = open(file=\"mgrs_world.txt\").read().splitlines()\n",
+    "mgrs_tiles.sort(key=lambda m: m[2])  # sort by latitudinal band from South to North"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "83ea0ca1-b6e3-4d6f-9f27-b1cd806c235e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Using bfloat16 Automatic Mixed Precision (AMP)\n",
+      "GPU available: True (cuda), used: True\n",
+      "TPU available: False, using: 0 TPU cores\n",
+      "IPU available: False, using: 0 IPUs\n",
+      "HPU available: False, using: 0 HPUs\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Setup trainer and load model weights from checkpoint\n",
+    "# !aws s3 cp s3://clay-model-ckpt/v0/clay-small-70MT-1100T-10E.ckpt checkpoints/\n",
+    "trainer = L.Trainer(precision=\"bf16-mixed\", logger=False)\n",
+    "model: L.LightningModule = CLAYModule.load_from_checkpoint(\n",
+    "    checkpoint_path=\"checkpoints/clay-small-70MT-1100T-10E.ckpt\"\n",
```

Comment:
Should be possible to load the checkpoint directly from HuggingFace now, instead of manually downloading from the s3 bucket, as mentioned at https://github.com/Clay-foundation/model/pull/116#discussion_r1451812210

```suggestion
    "trainer = L.Trainer(precision=\"bf16-mixed\", logger=False)\n",
    "model: L.LightningModule = CLAYModule.load_from_checkpoint(\n",
    "    checkpoint_path=\"https://huggingface.co/made-with-clay/Clay/resolve/main/Clay_v0.1_epoch-24_val-loss-0.46.ckpt\"\n",
```